### DISCUSSION TITLE
What about 4 channels oscilloscope?

### DISCUSSION THREAD

**Discussion by polex** [fhdm-dev/scoppy#110] [Q&A]
That project is great.
I plan to use the oscilloscope as a base for carburators syncronization tools which is used in motorcycle engine to balance the aspiration of each carburator at idle.
Typical tools measure the average vacuuum on each carburator. Instead a tool showing the vacum "waveform" may be helpful to identify potential issue on the engine. 
I already found this for 300€:
https://www.healtech-electronics.com/products/est/
but I like something more flexible: i.e. monitoring even the extra voltage produced by spark plug coils.
I identified ADP5111 as pression sensors to be used for carburators syncronization.
I may workaround the limitation of channels by measuring 2 signals among the 6:
- 4 pression sensor (1 for each cylinder)
- 2 spark plug coils
Why we do not use all the 4 channels available on Raspberry pico?

**Comment by fhdm-dev** [comment]
Hi,

The Pico only gives access to 3 of the 4 ADCs of the RP2040 but that still begs the question of why all of the 3 available ADCs are not used. 

It is simply that when I first developed the app it would have required extra time/effort to add support for the extra channel (expecially trying to fit the extra controls etc on the already busy user interface) and the app had already taken much longer to develop than I expected.

There's no technical reason why the 3rd channel can't be supported (except that it might reduce the waveform update rate too much on some phones) so it's possible that I might add support for it sometime in the future.

**Reply by polex** [reply]
Oppps, I did not realized that GPIO29/ADC3 was not available on the pico board.
Nevertheless, having to much waveform on small screen may be confusing.
I may easily use a multiplexer on the 2nd channel for selecting different waveform.

Thank you very much for developing such smart tool.
